### DISCUSSION TITLE
Timecodes in Deepgram results

### DISCUSSION THREAD

**Discussion by Jupitersbay** [deepgram/community#477] [General help] [answered]
Hello,
I was not lazy, searched the whole afternoon for a way to display timecodes in the results, but haven't found a solution. 
How can I dispaly timecodes in the results? (best: csv-or txt-file result)
Thank you

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jkroll-deepgram** [comment]
Hi @Jupitersbay, can you please elaborate on how you'd like to use timecodes in your transcription? Would you like to retain timecode information at the word-level, similar to our timestamps? This isn't a feature we currently offer by default, but it's possible that you could use Deepgram's timestamps with additional logic to preserve timecodes. Is there a certain audio file format you're working with that includes timecode metadata?

Also note that our API always returns JSON responses, though those can be post-processed into other file formats such as csv or txt.

**Comment by Jupitersbay** [comment]
hi Julia,

thanks for answering.

we need timecodes like in subtitles. timecodes for the dialogues so we can
create texte between the dialogues for the audiodescription.

In my first test of deepgram in the browser-console the german
translation and textunderstanding was excellent and the sentences were
combined in paragraphs. Actually it would be great to have timecodes for
text, that belongs together - like with subtitles in .srt-files. Is
something like that doable?

I understood that there could be some postprocessing done for the
JSON-files, but the second test, done from Pythonn, that gave JSON files
were not as excellent since it gave many single words instead of
paragraphs. Probably a problem with the options.

Can you help?

Marlon

MARLON ROSENTHAL
sprecherprofi | german voice-over [image: Logo]
<https://www.sprecherprofi.de/>

sprecherprofi
Binger Str. 83
DE-14197 Berlin
m: +49 163 6002666
e: ***@***.***

www.sprecherprofi.de [image: Facebook icon]
<https://www.facebook.com/MarlonRosenthal>   [image: LinkedIn icon]
<https://de.linkedin.com/in/marlon-rosenthal-18383b21>

[image: Banner] <https://www.sprecherprofi.de/>


Am Mo., 4. Dez. 2023 um 19:56 Uhr schrieb Julia Kroll <
***@***.***>:

**Reply by jkroll-deepgram** [reply, marked as answer]
Hi Marlon, we do offer SRT and WebVTT captioning outputs through our JavaScript and Python SDKs. They associate a timestamp range with a text phrase.

Docs: [Automatically Generating WebVTT and SRT Captions](https://developers.deepgram.com/docs/automatically-generating-webvtt-and-srt-captions)
Standalone libraries that integrate with our API for [JavaScript](https://github.com/deepgram/deepgram-js-captions), [Python](https://github.com/deepgram/deepgram-python-captions).

These libraries can give you a transcript output with timestamps, like:
```
1
00:00:00,080 --> 00:00:03,220
Yeah. As as much as, it's worth celebrating,

2
00:00:04,400 --> 00:00:07,859
the first, spacewalk, with an all female team,

3
00:00:08,475 --> 00:00:10,715
I think many of us are looking forward
```

Does that look like the kind of output you're looking to get?

**Comment by Jupitersbay** [comment]
Hi Julia,

thats super! I can't believe it. Must have been blind. Today I tried again
on your playground area all the parameters. I guess, there is no timestamp
parameter in the playground version, correct?

I coded the JSON version today in Python, but - of course - thats not what
we really are looking for.
So do you mind, giving me the python code that created the little beauty
you demonstrated above?
That would help a lot!

Thank you,

Marlon

MARLON ROSENTHAL
sprecherprofi | german voice-over [image: Logo]
<https://www.sprecherprofi.de/>

sprecherprofi
Binger Str. 83
DE-14197 Berlin
m: +49 163 6002666
e: ***@***.***

www.sprecherprofi.de [image: Facebook icon]
<https://www.facebook.com/MarlonRosenthal>   [image: LinkedIn icon]
<https://de.linkedin.com/in/marlon-rosenthal-18383b21>

[image: Banner] <https://www.sprecherprofi.de/>


Am Di., 5. Dez. 2023 um 21:05 Uhr schrieb Julia Kroll <
***@***.***>:

**Reply by jkroll-deepgram** [reply]
I'm glad to hear that this is the type of formatting that you're looking for!

Here's an example using our Python SDK:
```
from deepgram import Deepgram
deepgram = Deepgram("<your API key>")
response = deepgram.transcription.sync_prerecorded({"url": "https://static.deepgram.com/examples/Bueller-Life-moves-pretty-fast.wav"}, {'punctuate': True, 'smart_format': True, 'model': 'nova', 'utterances': True})
deepgram.extra.to_SRT(response)
```

That outputs:
```
1
00:00:05,519 --> 00:00:06,019
Yep.

2
00:00:07,094 --> 00:00:08,615
I said it before, and I'll say it

3
00:00:08,615 --> 00:00:09,115
again.

4
00:00:09,974 --> 00:00:11,514
Life moves pretty fast.

5
00:00:11,974 --> 00:00:13,734
You don't stop and look around once in

6
00:00:13,734 --> 00:00:15,701
a while. You could miss it
```

Depending on whether you want SRT or WebVTT, you can use `deepgram.extra.to_SRT(response)` or `deepgram.extra.to_WebVTT(response)`.

We don't offer the ability to export captions in the Playground, but you may be able to make use of sentences and paragraphs. You can see in `["results"]["channels"][0]["alternatives"][0]["paragraphs"]` that the response includes lists of paragraphs, which in turn includes lists of sentences. All the sentences and paragraphs have start- and end-timestamps. You could use those time ranges to create your own captions similar to the ones supported in the SDKs. Do note that Deepgram is primarily an API, so we do not offer a great deal of exporting customization through the Playground, since we expect users to typically work with the raw JSON.

I hope you can make these options work for your use case!

![Screenshot 2023-12-05 at 2 38 21 PM](https://github.com/deepgram/community/assets/135150417/e442abd8-8c0f-4cf8-a594-fea9a83ffb27)

**Reply by krezlic** [reply]
Hi, 

Looking at the possibility of time stamping the transcript with timecode in an associated audio file-  ( WAV / BWAV ) 
We want to feed in a recorded file and then output it with the timecode on the paragraphs to easily locate this within a video

Many thanks for your guidance 

Nick

**Reply by jkroll-deepgram** [reply]
@krezlic Hi Nick, as I mentioned earlier, Deepgram does provide timestamps at the word, sentence, and paragraph level. However, all of our timestamps start at time 0, which is the start of the audio file. You could then do some post-processing to reconcile your timecodes with the start time and duration of an audio file. For instance, if you have an audio file that starts at 10:48:43:12 (hours:minutes:seconds:frames), which we consider time 0, then a word that we mark as 6.96 (6 seconds, 960 milliseconds), and assuming 24 frames per second, would be 10:48:43:12 + 00:00:06:00 + (960/1000*24 = 23 frames) = 10:48:50:11.
